### PR TITLE
Plan `limits` system.

### PR DESCRIPTION
**Pull request by spolu**

### CODE REVIEW

File: front/lib/auth.ts [hunk 1]
```diff
@@ -297,3 +300,59 @@ export async function getAPIKey(
...
+    dataSources: {
+      count: DEFAULT_DATASOURCES_COUNT_LIMIT,
+      documents: {
+        count: DEFAULT_DATASOURCES_DOCUMENTS_COUNT_LIMIT,
+        sizeMb: DEFAULT_DATASOURCES_DOCUMENTS_SIZE_MB_LIMIT,
+      },
+    },
+  };
+
+  if (w.plan) {
```
Comment (lasryaric) [inline front/lib/auth.ts:319]: Can we use a little parse_payload here?
ChatGPT generated that:
```
{
  "$schema": "http://json-schema.org/draft-07/schema#",
  "definitions": {
    "LimitsType": {
      "type": "object",
      "properties": {
        "dataSources": {
          "type": "object",
          "properties": {
            "count": { "type": "number" },
            "documents": {
              "type": "object",
              "properties": {
                "count": { "type": "number" },
                "sizeMb": { "type": "number" }
              },
              "required": ["count", "sizeMb"],
              "additionalProperties": false
            }
          },
          "required": ["count", "documents"],
          "additionalProperties": false
        }
      },
      "required": ["dataSources"],
      "additionalProperties": false
    },
    "PlanType": {
      "type": "object",
      "properties": {
        "limits": { "$ref": "#/definitions/LimitsType" }
      },
      "required": ["limits"],
      "additionalProperties": false
    }
  }
}

```

I did not test it though.

Here is the prompt:
```
Can you generate the AJV JSONSchemaType for the following interface?
export type LimitsType = {
  dataSources: { count: number; documents: { count: number; sizeMb: number } };
};

export type PlanType = {
  limits: LimitsType;
};```

**Comment by spolu** [comment]
@lasryaric feel free to review of course as well if you get to it first :+1:

**Review by lasryaric — CHANGES_REQUESTED** [review]
Asking to change the payload parsing. Otherwise all good.

**Review by lasryaric — APPROVED** [review]
LGTM except the count() of document which can become quite slow on a big data source.
We discussed some potential solutions.

**Comment by spolu** [comment]
Proposed plan if needed:
- Do not pull count if the limit is infinite (likely correlated with large data sources)
- Store a total somewhere else and pull from it.

**Pull request merged**